pandas/core/groupby/grouper.py

```diff
@@ -3,7 +3,7 @@
 split-apply-combine paradigm.
 """
 
-from typing import Optional, Tuple
+from typing import Hashable, List, Optional, Tuple
```

Comment:
there are some places in these modules where `name` attrs are annotated as `str` or `Optional[str]` where Hashable would be more accurate I think